examples/experimental/dagster-components/dagster_components/impls/dbt_project.py

```diff
@@ -0,0 +1,41 @@
+from dagster._core.definitions.definitions_class import Definitions
+from dagster._utils import pushd
+from dagster_dbt import DbtCliResource, DbtProject, dbt_assets
+from dagster_embedded_elt.sling.resources import AssetExecutionContext
+from pydantic import TypeAdapter
+from typing_extensions import Self
+
+from dagster_components import Component, ComponentLoadContext
+from dagster_components.core.component_decl_builder import ComponentDeclNode, YamlComponentDecl
+
+
+class DbtProjectComponent(Component):
+    params_schema = DbtCliResource
+
+    def __init__(self, dbt_resource: DbtCliResource):
+        self.dbt_resource = dbt_resource
+
+    @classmethod
+    def registered_name(cls) -> str:
+        return "dbt_project"
+
+    @classmethod
+    def from_decl_node(cls, context: ComponentLoadContext, decl_node: ComponentDeclNode) -> Self:
+        assert isinstance(decl_node, YamlComponentDecl)
+
+        # all paths should be resolved relative to the directory we're in
```

Comment:
more color: this lets us configure the dbt resource with `project_dir: jaffle_shop` and have that be relative to the file where that yaml is defined.